dev_docs/key_concepts/persistable_state.mdx

```diff
@@ -0,0 +1,68 @@
+---
+id: kibDevDocsPersistableStateIntro
+slug: /kibana-dev-docs/persistable-state-intro
+title: Persistable State
+summary: Persitable state is a key concept to understand when building a Kibana plugin.
+date: 2021-02-02
+tags: ['kibana','dev', 'contributor', 'api docs']
+---
+
+ “Persistable state” is developer-defined state that supports being persisted by a plugin other than the one defining it. Persistable State needs to be serializable and the owner can/should provide utilities to migrate it, extract and inject references from/to it, as well as telemetry collection utilities.
+ 
+## Exposing state that can be persisted
+
+Any plugin that exposes state or configuration that another plugin might persist should implement `PersistableStateService`
+interface on their `setup` contract. This will allow plugins persisting the state to easily access migrations and other utilities.
+
+example: Data plugin allows you to generate filters. Those filters can be persisted by applications in their saved 
+objects or in the URL. In order to allow apps migrating the filters in case the structure changes in the future Data 
+plugin implements `PersistableStateService` on `data.query.filterManager`.
+
+## exposing state that can be persisted but is not owned by plugin exposing it (registry)
+
+Any plugin that owns collection of items whose state/configuration can be persisted should implement `PersistableStateService`
+interface on their `setup` contract as well as make each item in the collection implement `PersistableStateDefinition` interface.
+
+example: Embeddable plugin owns the registry of embeddable factories to which other plugins can register new embeddable factories. Dashboard plugin
+stores a bunch of embeddable panels input in its saved object and URL. Embeddable plugin setup contract implements `PersistableStateService`
+interface and each `EmbeddableFactory`  needs to implement `PersistableStateDefinition` interface.
+
+note: if your plugin doesn't expose the state and is thus the only one storing state but still owning the registry
+of items whose state it doesn't own then the items should still implement `PersistableStateDefinition` but registry
+plugin doesn't need to implement `PersistableStateService` interface.
+
+## storing persistable state as part of saved object
+
+Any plugin that stores any persistable state as part of their saved object should make sure that its saved object migration 
+and reference extraction and injection methods correctly use the correct `PersistableStateService` implementation.
+
+For migrations saved object definition `migration` key should be a function, which returns a list of migrations. `getAllMirations`
+function of `PersistableStateService` should be called in there to get an object with migrations which should be merged with
+any migrations provided by the plugin.
```

Comment:
```suggestion
If a saved object contains state from another plugin, the <DocLink id="kibCoreSavedObjectsPluginApi" section="iaf33fb81-e64a-11eb-8631-9d6e17a74ca4" text-"SavedObjectType.migrations"/> parameter should be a function, which returns a list of migrations. 
 These migration functions will need to merge the migration functions returned from <DocLInk id="" section="" text="PersistableStateServices.getAllMigrations"/> with any migrations provided by the plugin.
```

^^ I didn't finish the DocLinks, or verify they are correct, but hopefully you get the idea and can fill them in.